crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -183,3 +249,16 @@ pub struct RegularPolygon {
     pub sides: usize,
 }
 impl Primitive2d for RegularPolygon {}
+
+impl RegularPolygon {
+    /// Create a new `RegularPolygon`
+    /// from the radius of the circumcircle and number of sides
+    pub fn new(circumcircle_radius: f32, sides: usize) -> Self {
```

Comment:
This needs an assert that the radius is positive, and a `Panics` section in the docs.

Reply:
Should be solved in the latest commit 👍